apps/bot/src/libs/modules/socket-client/base-socket.module.ts

```diff
@@ -0,0 +1,57 @@
+import { io, type Socket } from "socket.io-client";
+
+import {
+	type ClientToServerEvents,
+	type ServerToClientEvents,
+	type SocketReservedEvents,
+} from "./libs/types/types.js";
+
+type Listener<
+	K extends keyof ServerToClientEvents | keyof SocketReservedEvents,
+> = K extends keyof SocketReservedEvents
+	? SocketReservedEvents[K]
+	: K extends keyof ServerToClientEvents
+		? ServerToClientEvents[K]
+		: never;
```

Comment:
```suggestion
type Events = ServerToClientEvents & SocketReservedEvents;
type Listener<K extends keyof Events> = Events[K];
```
can sth like this work? probably there will be adjustments, but the main point is to simplify the readability, as this type is too complicated for now

Reply:
we have also :
```ts
type SocketReservedEvents = {
	connect: () => void;
	connect_error: (error: Error) => void;
	disconnect: (reason: string) => void;
};
export { type SocketReservedEvents };
```
i leave for now as:
```ts
type EventKeys = keyof ServerToClientEvents | keyof SocketReservedEvents;
type Listener<K extends EventKeys> = K extends keyof SocketReservedEvents
	? SocketReservedEvents[K]
	: K extends keyof ServerToClientEvents
		? ServerToClientEvents[K]
		: never;
```